### ISSUE TITLE
AlsoCouldBeApply: false positive when all statements are not `it`-started expressions

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#5375]
## Steps to Reproduce
```kotlin
@Test
fun `does not report when all statements are not 'it'-started expressions`() {
    val code = """
        fun test(foo: Foo) {
            foo.also {
                it.bar()
                println(it)
                it.baz()
            }
        }

        class Foo {
            fun bar() {}
            fun baz() {}
        }
    """.trimIndent()
    assertThat(subject.compileAndLint(code)).isEmpty()
}
```
